### PR TITLE
Switch documentation from guzzle-sphinx-theme to furo

### PR DESCRIPTION
**Pull request by behackl**
## Changelog / Overview

This PR proposes to change our default documentation theme from `guzzle-sphinx-theme` (which isn't actively developed any longer) to `furo` (which is overall much nicer, actively developed, better customizable (=less workarounds)) Sphinx base theme.

Bonus features:
- Slight color changes (once again :-)) for better contrast
- *dark mode* (yep!)

## Motivation
See above.

## Testing Status

Let's see whether the RTD pipeline will deploy this PR.

## Further Comments

This is just a proposal, it wasn't too much effort to build this -- but I felt like playing around with it a bit. If the community doesn't like it, we can simply not merge this PR. However, independent of whether you like the look and feel of this theme (I particularly like the fact that the light theme is now closer to our landing page side, https://www.manim.community): switching to a maintained Sphinx theme would be a good idea.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [x] The PR title is descriptive enough
- [x] The PR is labeled correctly
- [x] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by WampyCakes** [comment]
I am generally a fan of furo, but sometimes I think it just doesn't fit right. In particular, I think relatively small docs look best with this theme. That being said, this looks _okay_. I noticed a bit of buggy behavior (like clicking any of the dropdowns in the lefthand sidebar). I also think that if we used this theme, we would **have** to update some CSS styling throughout the entire site if possible.

Also, is dark theme specified by the user or the site as a whole? I'm not sure the user can choose. If so, I missed it.

**Comment by behackl** [comment]
> I am generally a fan of furo, but sometimes I think it just doesn't fit right. In particular, I think relatively small docs look best with this theme. That being said, this looks _okay_.

Taste differs. I like this theme more than the current one and think that the mininalistic design fits very well to the documentation.

> I noticed a bit of buggy behavior (like clicking any of the dropdowns in the lefthand sidebar). 

What exactly does not work or has weird behavior for you? I can't really find any particular problems with the dropdowns.

> I also think that if we used this theme, we would **have** to update some CSS styling throughout the entire site if possible.

Sure it is possible, I already ship this with some custom CSS. Please be more specific about what you would like to see updated.

> Also, is dark theme specified by the user or the site as a whole? I'm not sure the user can choose. If so, I missed it.

The user. The site uses the light/dark theme preference as supplied by the browser. Depending on your OS and browser combination, changing it might require you to change the preferred appearance in the OS system settings.

**Review by jsonvillanueva — APPROVED** [review]
image
Dark theme looks great btw

File: docs/source/_static/custom.css [hunk 1]
```diff
@@ -119,68 +28,25 @@ div.seealso {
...
-.manim-example .copybtn {
+.admonition-manim-example .copybtn {
     margin-right: 6px;
     font-size: 18px;
 }
 
-.manim-example .copybtn:hover {
+.admonition-manim-example .copybtn:hover {
     cursor: pointer;
 }
```
Comment (naveen521kk) [inline docs/source/_static/custom.css:52]: add a new line at end of file

File: docs/source/conf.py [hunk 2]
```diff
@@ -85,25 +85,37 @@
...
+    'dark_css_variables': {
+        'color-content-foreground': '#ffffffd9',
+        'color-background-primary': '#131416',
+        'color-background-border': '#303335',
+        'color-sidebar-background': '#1a1c1e',
+        'color-brand-content': '#2196f3',
+        'color-brand-primary': '#87c2a5',
+    }
+}
+html_title = "Manim Community"
```
Comment (naveen521kk) [inline docs/source/conf.py:117]: Maybe change it to like `Manim Documentation` like how it was previously.

Comment (jsonvillanueva) [reply docs/source/conf.py:117]: If it's possible using an f-string, I would like the version number to be in the title as well similar to how Python does it:
https://docs.python.org/3/

**Comment by WampyCakes** [comment]
> > I noticed a bit of buggy behavior (like clicking any of the dropdowns in the lefthand sidebar).
> 
> What exactly does not work or has weird behavior for you? I can't really find any particular problems with the dropdowns.

image

 
> > I also think that if we used this theme, we would **have** to update some CSS styling throughout the entire site if possible.
> 
> Sure it is possible, I already ship this with some custom CSS. Please be more specific about what you would like to see updated.

I think a lot of the text is a bit too small. Here are two examples. In the pictures, they seem fine. But in comparison to the text surrounding them on the page, they seem arbitrarily small. I'm specifically looking at headers like "Tip" and "ExampleManimCELogo." The text under "Tip" is also too small along with the code and references. I think besides being out-of-place, it would be hard to read for some people.
image
image

Hyperlinks need updated throughout the whole site to fit in with our color scheme.
image

And then for all the reference pages ([like this one](https://manimce--1251.org.readthedocs.build/en/1251/reference/manim.mobject.geometry.Circle.html#manim.mobject.geometry.Circle)), I think they would benefit from a clearer separation of content. I find it a bit difficult to follow the hierarchy.

I think it would look nice if we had one piece that contrasted every page a bit more. Perhaps a slightly bigger contrast between the sidebar background and the page background would look good. Just a thought

With some CSS changes though, I could really get behind this. I like that furo is also more maintained.

> > Also, is dark theme specified by the user or the site as a whole? I'm not sure the user can choose. If so, I missed it.
> 
> The user. The site uses the light/dark theme preference as supplied by the browser. Depending on your OS and browser combination, changing it might require you to change the preferred appearance in the OS system settings.

That's too bad. You oughta be able to specify in the site :/ Sometimes I prefer light _or_ dark themes depending on what it is

**Comment by PhotonSpheres** [comment]
I really like that new doc style and I appreciate the dark mode! Stuff like the pic below should be fixed I guess since the text on the arrows is quite hard to read now. There shouldn't be a lot of those examples as far as I'm aware.

screen

**Comment by kolibril13** [comment]
image
https://docs.manim.community/en/latest/reference/manim.mobject.mobject.Mobject.html?highlight=add_updater#manim.mobject.mobject.Mobject.add_updater
image
https://manimce--1251.org.readthedocs.build/en/1251/reference/manim.mobject.mobject.Mobject.html#manim.mobject.mobject.Mobject.add_updater
I think the current docs theme looks a big "tighter" , and the greyish background makes the new one more difficult to read.
In general, I am ok with the new theme, but overall I think it still needs a lot of fine-tuning. 
One idea that I would have to the images that I posted:
Why don't we make  the background white and switch the highlight bar that is currently yellow to community white/greyish?

**Comment by behackl** [comment]
If the general opinion is more a "it would be okay to switch" than a "changing the theme is a general improvement of our documentation", then I'll rather close this PR and stay with how things are currently. If you would like to do that and think it would make sense, we can do a Discord poll or so.

I do appreciate all the feedback I've received so far! For many things reported here I don't see a clear path forward -- and I know that discussing over design-related questions is a fairly subjective matter; it's unlikely that we'll arrive at a version that everyone is 100% happy with. I'll try to address the raised issues one by one.

@WampyCakes: I can't reproduce the navbar behavior. What browser are you using, and how large is your viewport? (For reference: this is how it looks for me in Chrome, Edge, Firefox (and I've also tested Safari, but not on the screenshot):
image
)

> I think a lot of the text is a bit too small.

I don't really have that impression, I actually really like the font sizes right now. I can try to make a suggestion with adjusted sizes, but I fear that increasing it too much will quickly make the theme look less clean, which I would not want.

> But in comparison to the text surrounding them on the page, they seem arbitrarily small.

Letting the admonition blocks have a slightly smaller font size (80% of the normal font size) again helps to let the blocks appear a bit more separated and clean, compare:

image
image

I personally prefer having the block in a smaller font size.

> it would be hard to read for some people.

The smallest font size we have is still larger than 12px, which is the smallest recommended size for small text.

> Hyperlinks need updated throughout the whole site to fit in with our color scheme.

I'm open for suggestions, I couldn't find a good color which (a) is close to our color scheme (b) has a sufficiently high contrast to the background and (c) has a sufficiently high contrast to surrounding text.

> And then for all the reference pages (like this one), I think they would benefit from a clearer separation of content. I find it a bit difficult to follow the hierarchy.

Agreed, although I think that the [current version](https://docs.manim.community/en/stable/reference/manim.mobject.geometry.Circle.html#manim.mobject.geometry.Circle) of the documentation suffers from the same problem. We should look into introducing proper sections for these parts of the manual; I don't really see that as part of this PR though.

> I think it would look nice if we had one piece that contrasted every page a bit more. Perhaps a slightly bigger contrast between the sidebar background and the page background would look good. Just a thought

Increasing the contrast between sidebar and background also makes it more difficult to use the same colors/fonts in both of them.

> That's too bad. You oughta be able to specify in the site :/ Sometimes I prefer light or dark themes depending on what it is

I agree, but I don't intend to add a manual switch to this PR. It seems like it's not a straightforward thing to add, unfortunately -- and furo itself doesn't support it.

> Stuff like the pic below should be fixed I guess since the text on the arrows is quite hard to read now.

Agreed. Who has access to the Google Drive this thing is currently hosted on; do you remember @kolibril13? Adding a white background with 50% opacity should resolve the problem; I could also do that manually via adding some CSS but would prefer if we could change it in the diagram.

> I think the current docs theme looks a big "tighter" ,

Do you think that making it less tight is better or worse? A larger line height usually helps with making things easier to read.

> and the greyish background makes the new one more difficult to read.

We do loose a bit for the contrast ratio (12.2 in the current docs for text on background vs. 10.8 in the proposed version), but in terms of accessibility this is still pretty good. I think matching the design of our landing page / our logo makes this worth to have.

> but overall I think it still needs a lot of fine-tuning.

I'm sorry, but you need to be more specific about what exactly should be fine-tuned in your opinion.

**Comment by behackl** [comment]
Any further comments or feedback on this?

**Comment by RickyC0626** [comment]
> Any further comments or feedback on this?

I'm a fan of not being blinded while reading documentation.

**Review by friedkeenan — APPROVED** [review]
As it is currently, I think it's an improvement over the guzzle theme. It could probably be improved a bit more with what was discussed but I don't think any of that's worth blocking over.

**Review by naveen521kk — APPROVED** [review]
sgtm, other than
https://github.com/ManimCommunity/manim/pull/1251/files#r607062327
https://github.com/ManimCommunity/manim/pull/1251/files#r607062752

**Comment by WampyCakes** [comment]
> If the general opinion is more a "it would be okay to switch" than a "changing the theme is a general improvement of our documentation", then I'd rather close this PR and stay with how things are currently. 

My opinion is that this is a general improvement, but fine tuning is necessary for this theme to be better. My opinion is that changing the theme would be an improvement, to furo or a different theme.

> @WampyCakes: I can't reproduce the navbar behavior. What browser are you using, and how large is your viewport?

I am using Firefox 68 on a normal size screen at 100% zoom, browser taking up the whole screen (not fullscreen). Yes, it's a bit old, but we can safely expect that we will have users with much older browsers than that, and most websites work flawlessly for me on this version.

It does work for me on Firefox 77, but that is on a different computer entirely.

> Letting the admonition blocks have a slightly smaller font size (80% of the normal font size) again helps to let the blocks appear a bit more separated and clean

I disagree with this. I still think it's too small to expect everyone to comfortably read (I find it harder (but possible) myself already). I particularly care about the size of the code tho. The code is a lot smaller and it seems out-of-place because it's not like a one word header.

> > Hyperlinks need updated throughout the whole site to fit in with our color scheme.
> 
> I'm open for suggestions, I couldn't find a good color which (a) is close to our color scheme (b) has a sufficiently high contrast to the background and (c) has a sufficiently high contrast to surrounding text.

I would recommend trying this. Set the background of hyperlinks to the same background color of the page (instead of white) and then set the text color to manim's `LOGO_RED`. It should blend in while making the hyperlink pop out. Also might look better if `text-decoration` is set to `none` to remove the underline.

> > And then for all the reference pages (like this one), I think they would benefit from a clearer separation of content. I find it a bit difficult to follow the hierarchy.
> 
> Agreed, although I think that the [current version](https://docs.manim.community/en/stable/reference/manim.mobject.geometry.Circle.html#manim.mobject.geometry.Circle) of the documentation suffers from the same problem. We should look into introducing proper sections for these parts of the manual; I don't really see that as part of this PR though.

Agreed on this. Though it _is_ a little worse with this theme imo so it should be a priority.

> I think matching the design of our landing page / our logo makes this worth to have.

This would be a really good touch to make our websites more coherent.

**Comment by WampyCakes** [comment]
Also, is it going to be possible to update the docs for our previous versions or just going forward?

(Outside the scope of this PR) Is there a way to alert users that they are viewing older documentation?

**Comment by behackl** [comment]
Regarding the navbar on older browsers: I've played around a bit and might have a working solution. Please try whether it looks better in the new version (as soon as the RTD pipeline finishes).

> I particularly care about the size of the code tho.

I agree, code size was too small. Adapted to the same size as in non-admonition blocks; and I think it still works.

> I would recommend trying this. Set the background of hyperlinks to the same background color of the page (instead of white) and then set the text color to manim's LOGO_RED. It should blend in while making the hyperlink pop out. Also might look better if text-decoration is set to none to remove the underline.

image
image


I'd be fine with that personally; the contrast for light mode isn't great though. The white background stems from some of the links being formatted as inline code, and inline code does need a little more help to be able to distinguish it easily (actually, I'd like to have a stronger contrast than there is currently for dark mode). I have a slight preference for keeping the text-decoration. :-)

> Also, is it going to be possible to update the docs for our previous versions or just going forward?

Just going forward. You can still see the first version of our doc theme when visiting `v0.1.0`. ;-)

> (Outside the scope of this PR) Is there a way to alert users that they are viewing older documentation?

I don't know a good strategy to do so, maybe RTD offers an easy way for doing that.

**Comment by jsonvillanueva** [comment]
> > (Outside the scope of this PR) Is there a way to alert users that they are viewing older documentation?
> 
> I don't know a good strategy to do so, maybe RTD offers an easy way for doing that.

You'd have to modify this in conf.py -- I've seen cloup/click do this in their source code

**Comment by WampyCakes** [comment]
> Regarding the navbar on older browsers: I've played around a bit and might have a working solution. Please try whether it looks better in the new version (as soon as the RTD pipeline finishes).

Same problem I'm afraid

**Comment by behackl** [comment]
> > Regarding the navbar on older browsers: I've played around a bit and might have a working solution. Please try whether it looks better in the new version (as soon as the RTD pipeline finishes).
> 
> Same problem I'm afraid

Are you sure that nothing was cached? Can you make sure that the `.sidebar-tree a.internal.reference` rule which sets `display: table-cell` is present for you for the links in the navbar?

Otherwise I actually can't reproduce it (looking at it in FF68 right now) this time and I'll need you to provide more debug information.

**Comment by kolibril13** [comment]
Regarding the grey background:
I'd definitely prefer a white background, as currently, the contrast is very low:
https://contrast-ratio.com/?utm_campaign=chrome_series_contrastratio_050417&utm_source=chromedev&utm_medium=yt-desc#%23D6644D-on-%23EDE7E3 (only 2.9 for grey/orange)
image
Further, the dropdown menu is I think a bit weird because it is super long:
image

> Agreed. Who has access to the Google Drive this thing is currently hosted on; do you remember @kolibril13? Adding a white background with 50% opacity should resolve the problem; I could also do that manually via adding some CSS but would prefer if we could change it in the diagram.

Yes, I can take care of that.

> Do you think that making it less tight is better or worse? A larger line-height usually helps with making things easier to read.

I think less tight is worse because it looks a bit looser, but I guess this subjective. When it makes it easier to read, then I am fine with the larger line-height.

I think also the reference can be a bit larger:
image

**Comment by kolibril13** [comment]
I've now made here a new document: https://app.diagrams.net/#G1aKyJTloYB97IhrzwqEENOu-WQyuVWMjM
And added it to the docs here: https://github.com/ManimCommunity/manim/pull/1289

**Comment by behackl** [comment]
> I'd definitely prefer a white background, as currently, the contrast is very low

I want to keep a background color close (if not identical) to our light logo background. I'm open for link color suggestions, but if we (i.e., the people currently participating in this discussion) can't reach consensus about it soon, I suggest to keep it as-is for now and deal with it in a follow-up PR.

> Further, the dropdown menu is I think a bit weird because it is super long:

Not much that I can do about that on this PR. It is as it is, because Manim's modules are structured as they currently are. 🤷

> I think also the reference can be a bit larger:

I'd like to keep font size (apart from the code) consistent with the remaining text (i.e., the title) in the example block.

**Comment by WampyCakes** [comment]
Three things:
1. Dropdowns are working now, must have been cached and gotten cleared by now. Good job!
2. I agree with at least trying a white background to see how it looks. I think white and black are both universal colors that we can contrast our logo colors against. If it would look better white (just to have some extra contrast since we don't have two shades of LOGO_WHITE), not worth leaving it similar/identical to our logo colors. This PR is a closer color scheme in general so that's an improvement.
3. I also agree with larger reference text size. If text size on the few elements that have been mentioned are going to be a sticking point, time for a poll I guess.

**Comment by behackl** [comment]
> Three things:
> 
> 1. Dropdowns are working now, must have been cached and gotten cleared by now. Good job!

Great!

> 2. I agree with at least trying a white background to see how it looks. 

I did play around quite a bit with colors, but didn't find a satisfying combination for a purely white background. Feel free to play around yourself, but I won't invest more time into it; at least not for this PR. I feel that the current choice is acceptable, and fine-tuning can still happen in follow-up PRs.

> 3. I also agree with larger reference text size. If text size on the few elements that have been mentioned are going to be a sticking point, time for a poll I guess.

Font size increased via 12bcf6e8, it doesn't hurt to have it in the same size as the code directly above.

File: docs/source/conf.py (hunk 2 above)
Comment (behackl) [reply docs/source/conf.py]: Suggestion: 
image

Whenever I announce a new version, I announce it as "Manim Community vX.Y.Z", and I actually think it makes more sense to underline that this page concerns the community edition in version X.Y.Z, than to point out that this is the documentation.

Pushed via 751accea.

**Comment by naveen521kk** [comment]
The docs looks super nice 👍, thanks for this PR.

**Pull request merged**